### PR TITLE
Continue work on C++ packaging

### PR DESCRIPTION
**Pull request by jonsimantov**
After building packaging tools, upload them into an artifact so the rest of the build can retrieve them.

Also contains basic versions of the iOS and Android prereqs and build scripts, and build jobs added to the workflow. Uploads the built iOS and Android SDK artifacts as well.

Desktop needs to wait until desktop workflow is moved into prereqs + build scripts.

### CODE REVIEW

**Comment by jonsimantov** [comment]
Hi Vimanyu,

Could you take a look at this PR? The manual workflow is currently passing with the most recent run: https://github.com/firebase/firebase-cpp-sdk/actions/runs/178862172

I want to get this merged to avoid monolithic PRs. As we switch from .sh to .py scripts for building I'll update these scripts in a subsequent CL.

File: .github/workflows/cpp-packaging.yml [hunk 1]
```diff
@@ -9,55 +9,90 @@ env:
...
+      run: sdk-src/build_scripts/ios/install_prereqs.sh
+    - name: build sdk
+      run: |
+        sdk-src/build_scripts/ios/build.sh firebase-cpp-sdk-ios-${{ matrix.architecture }} sdk-src ${{ matrix.architecture }}
+        tar -czf firebase-cpp-sdk-ios-${{ matrix.architecture }}.tgz firebase-cpp-sdk-ios-${{ matrix.architecture }}
+    - name: upload artifacts
+      uses: actions/upload-artifact@v2
+      with:
+        name: firebase-cpp-sdk-ios-${{ matrix.architecture }}
+        path: firebase-cpp-sdk-ios-${{ matrix.architecture }}.tgz
```
[the remark this answers is not in the document]
Comment (jonsimantov) [reply .github/workflows/cpp-packaging.yml:77]: AFAIK these artifacts are local to the workflow? So I don't think they need versioning. Later on in the workflow, all of these will be joined together into a single binary SDK artifact, and *that* will be versioned.

File: .github/workflows/cpp-packaging.yml [hunk 2]
```diff
@@ -9,55 +9,90 @@ env:
...
-      - name: install demumble in /tmp/demumble
-        run: cp -af ./demumble /tmp/demumble
-        working-directory: ./demumble-src
+        run: |
+          cd demumble-src
+          cmake .
+          cmake --build .
+          python demumble_test.py
+          cd -
+          mkdir -p packaging-tools
```
[the remark this answers is not in the document]
Comment (jonsimantov) [reply .github/workflows/cpp-packaging.yml:45]: I don't think so, if there are any issues then the script won't get this far.

File: .github/workflows/cpp-packaging.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (DellaBitta) [reply .github/workflows/cpp-packaging.yml:77]: Yes, can confirm - they're scoped by the workflow.

**Pull request merged**